### PR TITLE
Fix DBT Required Field Fixes

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

Fix DBT Required Field Fixes

#
### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

**Review by nahuelverdugo — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 1]
```diff
@@ -369,10 +367,11 @@ def yield_data_models(self, dbt_files: DbtFiles) -> Iterable[DataModelLink]:
                         ),
                         datamodel=DataModel(
                             modelType=ModelType.DBT,
-                            description=manifest_node.get("description")
-                            if manifest_node.get("description")
+                            description=manifest_node.get("description"),
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:370]: what will happen if description is not there

Comment (harshach) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:370]: @ulixius9 should we do this ```manifest_node.get("description", {})```

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py [hunk 2]
```diff
@@ -369,10 +367,11 @@ def yield_data_models(self, dbt_files: DbtFiles) -> Iterable[DataModelLink]:
...
+                            path=f"{manifest_node['root_path']}/{manifest_node['original_file_path']}"
```
Comment (harshach) [inline ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:371]: IMO if we take this out here and put as if else block , would be easy to read

**Review by harshach — CHANGES_REQUESTED** [review]
@ulixius9 added minor comments. PTAL

**Comment by harshach** [comment]
@ulixius9 please look at the specification here https://schemas.getdbt.com/dbt/manifest/v8/index.html except the ones that are mentioned as "required" rest of the keys might not be present. 
Lets look at 1.3.x and 1.4 specifications and lets handle the extraction of data accordingly

**Comment by harshach** [comment]
Also lets look at this library https://github.com/yu-iskw/dbt-artifacts-parser
cc @OnkarVO7 @pmbrull

**Comment by OnkarVO7** [comment]
> Also lets look at this library https://github.com/yu-iskw/dbt-artifacts-parser cc @OnkarVO7 @pmbrull

@harshach created a issue [here ](https://github.com/open-metadata/OpenMetadata/issues/9976) for the same. Will integrate it for `1.0`

**Comment by harshach** [comment]
@OnkarVO7 we can't wait for this to be 1.0. Lets start working on it right away.

**Comment by OnkarVO7** [comment]
Got it @harshach will complete the integration for 0.13.2

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py (hunk 1 above)
Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:370]: @harshach description is of type [markdown](https://github.com/open-metadata/OpenMetadata/blob/8b45b639fd59abaf95baf55c3a46304bd10859eb/openmetadata-spec/src/main/resources/json/schema/entity/data/table.json#L767) and is an optional field so it's okay for description to be `None`

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py (hunk 2 above)
Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:371]: Sure, will do that

File: ingestion/src/metadata/ingestion/source/database/dbt/metadata.py (hunk 1 above)
Comment (harshach) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:376]: @ulixius9 talking about manifest_node.get("description") , description is an optional element from dbt manifest schema. Which means we may not have description as the key in manifest file which will throw up an error from get method saying key not found

Comment (ulixius9) [reply ingestion/src/metadata/ingestion/source/database/dbt/metadata.py:378]: @harshach  Yes the `description` is an optional field in manifest and `manifest_node.get("description")` will return `None` if the key `description` is not found 


here is an example 
image

**Comment by ulixius9** [comment]
Have made changes according to [this](https://schemas.getdbt.com/dbt/manifest/v8/index.html) specification

**Review by OnkarVO7 — APPROVED** [review]

**Comment by pmbrull** [comment]
these changes have been added in https://github.com/open-metadata/OpenMetadata/pull/9982. Closing this one

**Pull request closed without merging**